### PR TITLE
Refactor user actions

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
This PR extracts the code to parse middleware configuration vars,
and inlines methods previously in `utils/middleware`, as brought up by https://github.com/opentripplanner/otp-react-redux/pull/224#discussion_r492961846 and https://github.com/opentripplanner/otp-react-redux/pull/224#discussion_r497508647.

### CODE REVIEW

File: lib/actions/user.js [hunk 1]
```diff
@@ -28,102 +24,135 @@ function createNewUser (auth0User) {
...
 /**
- * Fetches the saved/monitored trips for a user.
- * We use the accessToken to fetch the data regardless of
- * whether the process to populate state.user is completed or not.
+ * Helper function that
+ * - extracts key variables from the state and passes them to the code to execute:
+ *     - apiKey and apiBaseUrl from state.otp.config.persistence.otp_middleware,
+ *     - accessToken and loggedIUnUser from state.user.
+ * - checks that otp_middleware is set, and throws an error if not.
+ * @param functionToExecute the code to execute, with parameters (dispatch, arguments)
```
Comment (evansiroky) [inline lib/actions/user.js:32]: Please note that `functionToExecute` can be a function that can be awaited upon.

**Review by evansiroky — APPROVED** [review]
Just one comment about a comment

**Review by landonreed — CHANGES_REQUESTED** [review]
This approach is far too clever. I really disagree with how the `executeWithMiddleware` method body is structured and how it is invoked throughout `user.js`.

**Comment by binh-dam-ibigroup** [comment]
> This approach is far too clever. I really disagree with how the `executeWithMiddleware` method body is structured and how it is invoked throughout `user.js`.

It highlights the core logic for these actions and hides repeat code to extract and check common variables.

**Review by evansiroky — CHANGES_REQUESTED** [review]
I'm going to have to change my review to request changes. I kind of agree with Landon in that this introduces a bit too much abstraction/overhead. I think the `executeWithMiddleware` shouldn't exist and instead it should simply be a util function that makes sure the middleware config exists and if so returns the desired middleware data given the getState method.

**Pull request merged**